src/include/catalog/catalog_accessor.h

```diff
@@ -397,6 +398,20 @@ class EXPORT CatalogAccessor {
    */
   common::ManagedPointer<transaction::TransactionContext> GetTxn() const { return txn_; }
 
+  /**
+   * Registers a temporary table in this accessor
+   * @warning For use in the execution engine only, such that cached modules function correctly
```

Comment:
Can we add a little more docs what registering means in this case? Why isn't there a corresponding unregister? Is there a bit more context for that warning?

Reply:
I can add some documentation as well. I was actually avoiding it deliberately to minimize the amount of stuff I was touching on the PR, but honestly I love documenting things so this would not be a issue for me.